src/translate.py

```diff
@@ -3253,6 +3253,21 @@ def handle_rlwinm(
 
     return BinaryOp.int(left=source, op="&", right=Literal(mask))
 
+def handle_loadx(source: Expression, type: Type) -> Expression:
+    size = type.get_size_bytes()
+    assert size is not None
+
+    # rD, rA, rB
+    if source.raw_arg(1) == Register('r0'):
+        # rA is 0, thus load from rB only
+        mem_addr = source.raw_arg(2)
+    else:
+        summed = BinaryOp.u32(source.reg(1), "+", source.reg(2))
+        source.regs[source.raw_arg(0)] = summed
```

Comment:
This should use an EvalOnceExpr, or else the + expression can be duplicated in multiple places. Thus, it needs to use the set_reg helper and this assignment be refactored out into the big translation loop somehow

Reply:
Thanks for the speedy reply. I'm seeing `set_reg` as part of the `translate_node_body` function -- does that need to be factored out, or am I missing something?

Reply:
I've moved the temporary register assignment into the translation loop (`2d4ca301c4cb638ddb98b2f446432a34d92b85ad`).

Reply:
Looks good! Can you use BinaryOp.intptr for the addition, btw? Same argument there as for the multiplication, except there's pointer addition in addition to addition of signed/unsigned.

Reply:
Gotcha. Fixed in `44dcae0`.